Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/ChangeEvent.swift

```diff
@@ -2,13 +2,22 @@ import MongoSwift
 
 /// The type of operation that occurred within a ChangeEvent.
 public enum OperationType: String, Codable {
-    case insert, delete, replace, update, unknown
+    case insert = "INSERT"
+    case delete = "DELETE"
+    case replace = "REPLACE"
+    case update = "UPDATE"
+    case unknown = "UNKNOWN"
 }
 
 /// Represents possible fields that a change stream response document can have.
 public struct ChangeEvent<DocumentT: Codable>: Codable, Hashable {
+    enum CodingKeys: String, CodingKey {
+        case id = "_id", operationType, fullDocument, ns
+        case documentKey, updateDescription, hasUncommittedWrites
+    }
+
     /// Metadata related to the operation.
-    public let id: Document
+    public let id: AnyBSONValue
```

Comment:
[q] why did this change? this is going to always be a document

Reply:
Is it? You can't have a BSON String or Integer ID?